commands/daemonize/service.go

```diff
@@ -0,0 +1,143 @@
+)
+
+// service has embedded daemon
+type service struct {
+	daemon.Daemon
+	args map[string]interface{}
+	log  *logrus.Logger
+}
+
+func (svc *service) Run() (string, error) {
+	logger := svc.log
+	err := setupSyslog(logger)
+	if err != nil {
+		return "Could not create syslog", err
+	}
+
+	// Find engine provider
+	engineName := svc.args["--engine"].(string)
+	engineProvider := extpoints.EngineProviders.Lookup(engineName)
+	if engineProvider == nil {
+		engineNames := extpoints.EngineProviders.Names()
+		return "Engine not found", fmt.Errorf("Must supply a valid engine. Supported Engines %v", engineNames)
+	}
+
+	// Create a temporary folder
+	tempPath := filepath.Join(os.TempDir(), slugid.Nice())
+	tempStorage, err := runtime.NewTemporaryStorage(tempPath)
+	runtimeEnvironment := &runtime.Environment{
+		Log:              logger,
+		TemporaryStorage: tempStorage,
+	}
+
+	// Initialize the engine
+	engine, err := engineProvider.NewEngine(extpoints.EngineOptions{
+		Environment: runtimeEnvironment,
+		Log:         logger.WithField("engine", engineName),
+	})
+	if err != nil {
+		return "Could not create engine", err
+	}
+
+	// TODO (garndt): Need to load up a real config in the future
+	config := &config.Config{
+		Credentials: struct {
+			AccessToken string `json:"accessToken"`
+			Certificate string `json:"certificate"`
+			ClientID    string `json:"clientId"`
+		}{
+			AccessToken: os.Getenv("TASKCLUSTER_ACCESS_TOKEN"),
+			Certificate: os.Getenv("TASKCLUSTER_CERTIFICATE"),
+			ClientID:    os.Getenv("TASKCLUSTER_CLIENT_ID"),
+		},
+		Taskcluster: struct {
+			Queue struct {
+				URL string `json:"url,omitempty"`
+			} `json:"queue,omitempty"`
+		}{
+			Queue: struct {
+				URL string `json:"url,omitempty"`
+			}{
+				URL: "https://queue.taskcluster.net/v1/",
+			},
+		},
+		Capacity:      5,
+		ProvisionerID: "test-dummy-provisioner",
+		WorkerType:    "dummy-worker-tc",
+		WorkerGroup:   "test-dummy-workers",
+		WorkerID:      "dummy-worker-tc",
+		QueueService: struct {
+			ExpirationOffset int `json:"expirationOffset"`
+		}{
+			ExpirationOffset: 300,
+		},
+		PollingInterval: 10,
+	}
+
+	l := logger.WithFields(logrus.Fields{
+		"workerID":      config.WorkerID,
+		"workerType":    config.WorkerType,
+		"workerGroup":   config.WorkerGroup,
+		"provisionerID": config.ProvisionerID,
+	})
+
+	w, err := worker.New(config, engine, runtimeEnvironment, l)
+	if err != nil {
+		return "Could not create worker", err
+	}
+
+	sigTerm := make(chan os.Signal, 1)
+	signal.Notify(sigTerm, syscall.SIGTERM)
+	go func() {
+		<-sigTerm
+		w.Stop()
+	}()
+	w.Start()
+	return "Worker successfully started", nil
+}
+
+// Manage by daemon commands or run the daemon
+func (svc *service) Manage() (string, error) {
+	action := svc.args["<action>"].(string)
+	// if received any kind of command, do it
+	switch action {
+	case "install":
+		if svc.args["--engine"] == nil {
+			return "Install failed", errors.New("--engine is mandatory for install command")
+		}
```

Comment:
You won't need this check if you specify each action separately in the usage (see my comment above).